File: crates/aipo-bench/comparisons/lua/fields.lua
local n = tonumber(arg[1])
local Box = {}
Box.__index = Box

function Box.advance(self, i)
  self.a = (self.b + i) % 1000
  self.b = (self.c + self.a) % 1000
  self.c = (self.d + self.b) % 1000
  self.d = (self.e + self.c) % 1000
  self.e = (self.f + self.d) % 1000
  self.f = (self.a + 1) % 1000
end

function Box.score(self)
  return self.a + 2 * self.b + 3 * self.c + 4 * self.d + 5 * self.e + 6 * self.f
end

local box = setmetatable({ a = 0, b = 0, c = 0, d = 0, e = 0, f = 0 }, Box)
local i = 0
local total = 0
while i < n do
  box:advance(i)
  total = total + box:score()
  i = i + 1
end
print("checksum:" .. total)

File: crates/aipo-bench/comparisons/javascript/fields.mjs
const n = Number(process.argv[2]);

class Box {
  constructor() {
    this.a = 0;
    this.b = 0;
    this.c = 0;
    this.d = 0;
    this.e = 0;
    this.f = 0;
  }

  advance(i) {
    this.a = (this.b + i) % 1000;
    this.b = (this.c + this.a) % 1000;
    this.c = (this.d + this.b) % 1000;
    this.d = (this.e + this.c) % 1000;
    this.e = (this.f + this.d) % 1000;
    this.f = (this.a + 1) % 1000;
  }

  score() {
    return this.a + 2 * this.b + 3 * this.c + 4 * this.d + 5 * this.e + 6 * this.f;
  }
}

const box = new Box();
let i = 0;
let total = 0;
while (i < n) {
  box.advance(i);
  total += box.score();
  i += 1;
}
console.log(`checksum:${total}`);
